crnk-core/src/main/java/io/crnk/core/utils/Optional.java

```diff
@@ -3,6 +3,7 @@
 import java.util.NoSuchElementException;
 import java.util.Objects;
 
+@Deprecated
```

Comment:
let me first prepare all the replacements before deprecating some api, otherwise just more questions will pop up...

looks everything great otherwise.